idseq_pipeline/commands/curate_accession2taxid.py

```diff
@@ -0,0 +1,102 @@
+                        dbids[s.group(1)] = 1
+    print "Read the mapping file and select ..."
+    # Read the accession2taxid mapping files
+    outf = open(output_mapping_file, 'wb')
+    lines = 0
+    for mapping_file in mapping_files:
+        with gzip.open(mapping_file, 'rb') as mapf:
+            for line in mapf:
+                fields = line.split("\t")
+                lines += 1
+                if lines % 100000 == 0:
+                    print "%f M lines. %s, %s" %  (lines/1000000.0, fields[0], fields[2])
+                if dbids.get(fields[0]):
+                    outf.write(line)
+    outf.close()
+
+def generate_accession2taxid_db(mapping_file, output_db_file, input_gzipped):
+    lines = 0
+    taxon_map = shelve.open(output_db_file)
+    if input_gzipped:
+        mapf = gzip.open(mapping_file, 'rb')
+    else:
+        mapf = open(mapping_file, 'rb')
+    for line in mapf:
+        fields = line.rstrip().split("\t")
+        if len(fields) == 4:
+            lines += 1
+            if lines % 100000 == 0:
+                print "%d lines. %s, %s" %  (lines, fields[0], fields[2])
+            accession_id = fields[0]
+            taxon_id = fields[2]
+            taxon_map[accession_id] = taxon_id
+    print "close the db file"
+    taxon_map.close()
+    mapf.close()
+
+class Curate_accession2taxid(Base):
+    def run(self):
+        from .common import * #TO DO: clean up the imports across this package
+
+        # Make work directory
+        dest_dir = os.path.join(DEST_DIR, 'accession2taxid')
+        execute_command("mkdir -p %s" % dest_dir)
+
+        # Retrieve the reference files
+        print "Retrieving references"
+        arguments = self.options
+        nt_file_local, nt_version_number = download_reference_locally_with_version_any_source_type(arguments['--nt_file'], dest_dir, dest_dir)
+        nr_file_local, nr_version_number = download_reference_locally_with_version_any_source_type(arguments['--nr_file'], dest_dir, dest_dir)
+        mapping_files_local = []
+        mapping_version_numbers = []
+        for f in arguments['--mapping_files'].split(","):
+            mapping_file_local, mapping_version_number = download_reference_locally_with_version_any_source_type(f, dest_dir, dest_dir)
+            mapping_files_local.append(mapping_file_local)
+            mapping_version_numbers.append(mapping_version_number) 
+        print "Reference download finished"
+
+        # Produce the output file
+        print "Curating accession2taxid"
+        output_mapping_file = os.path.join(DEST_DIR, 'curated_accession2taxid.txt')
+        curate_taxon_dict(nt_file_local, nr_file_local, mapping_files_local, output_mapping_file)
+        print "Curation finished"
+
+        # Convert to a berkeley db
+        print "Writing berkeley db"
+        output_db_file = os.path.join(DEST_DIR, 'curated_accession2taxid.db')
+        generate_accession2taxid_db(output_mapping_file, output_db_file, False)
```

Comment:
Do we need to erase the output file first?   In case it's a corrupt file from a failed prior run?   Not sure what the retry strategy will be.

Reply:
Yes - thanks!!